crates/wasmtime/src/runtime/vm/instance/allocator.rs

```diff
@@ -695,35 +677,69 @@ fn initialize_memories(instance: &mut Instance, module: &Module) -> Result<()> {
     // This call to `init_memory` notably implements all the bells and whistles
     // so errors only happen if an out-of-bounds segment is found, in which case
     // a trap is returned.
-    let ok = module.memory_initialization.init_memory(
-        instance,
-        InitMemory::Runtime {
-            memory_size_in_pages,
-            get_global_as_u64,
-        },
-        |instance, memory_index, init| {
+
+    struct InitMemoryRuntime<'a> {
```

Comment:
nitpicky suggestion, take it or leave it: I'd personally rename this `InitMemoryAtRunTime` and the other impl as `InitMemoryAtCompileTime`. I think it is a little more clear but maybe it is just me.

Reply:
I wasn't entirely happy with the names I picked and considered doing something like you suggest. So it's not hard to convince me on this one!